site/app/templates/error/NoAccessCourse.twig

```diff
@@ -1,18 +1,33 @@
 <div class="content">
-    You don't have access to this course. <br />
-    This is <b>{{course_name}}</b> for <b>{{semester}}</b>. <br/>
-    If you think this is a mistake,
-    {% if ability_to_readd %}
-        you may select below to temporarily re-add yourself to the course. <br/>
+    {% if not self_registration_type == 1 %}
```

Comment:
Magic values are *very* bad.  It would be better to use an enum for every reference to registration types.  (This applies elsewhere in this PR too)